server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/RequeueTest.java

```diff
@@ -0,0 +1,246 @@
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.mailets;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+import java.time.Duration;
+import java.util.Optional;
+
+import javax.mail.MessagingException;
+
+import org.apache.james.queue.api.MailQueueFactory;
+import org.apache.james.queue.api.MailQueueName;
+import org.apache.james.queue.api.RawMailQueueItemDecoratorFactory;
+import org.apache.james.queue.memory.MemoryMailQueueFactory;
+import org.apache.mailet.Mail;
+import org.apache.mailet.MailetConfig;
+import org.apache.mailet.base.test.FakeMail;
+import org.apache.mailet.base.test.FakeMailetConfig;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+public class RequeueTest {
+    private Mail mailSample;
+    private MemoryMailQueueFactory mailQueueFactory;
+    private MemoryMailQueueFactory.MemoryCacheableMailQueue spoolQueue;
+
+    @BeforeEach
+    void beforeEach() throws MessagingException {
+        mailSample = FakeMail.builder()
+            .name("mail1")
+            .sender("sender1@domain.tld")
+            .recipients("rcpt1@linagora.com")
+            .state("newState")
+            .build();
+        MemoryMailQueueFactory memoryMailQueueFactory = new MemoryMailQueueFactory(new RawMailQueueItemDecoratorFactory());
+        memoryMailQueueFactory.createQueue(MailQueueFactory.SPOOL);
+        spoolQueue = spy(memoryMailQueueFactory.getQueue(MailQueueFactory.SPOOL).orElseThrow());
+
+        mailQueueFactory = mock(MemoryMailQueueFactory.class);
+        when(mailQueueFactory.getQueue(MailQueueFactory.SPOOL)).thenReturn(Optional.of(spoolQueue));
+    }
+
+    private Requeue testee(MailetConfig mailetConfig) throws MessagingException {
+        Requeue requeue = new Requeue(mailQueueFactory);
+        requeue.init(mailetConfig);
+        return requeue;
+    }
+
+    @Test
+    void mailetShouldRequeueMail() throws MessagingException {
+        Requeue mailet = testee(FakeMailetConfig.builder()
+            .mailetName("Requeue")
+            .build());
+        mailet.service(mailSample);
+
+        ArgumentCaptor<Mail> mailCaptor = ArgumentCaptor.forClass(Mail.class);
+        Mockito.verify(spoolQueue).enQueue(mailCaptor.capture());
```

Comment:
Why do we need mocks if we have a memory implementation at hand?

I hate mocks.

Can we try rewrite this without mocks?